source/frontend/src/features/mapSideBar/management/tabs/activities/edit/models.ts

```diff
@@ -0,0 +1,224 @@
+import { fromApiPersonOrApiOrganization, IContactSearchResult } from '@/interfaces';
+import { ApiGen_Concepts_PropertyActivity } from '@/models/api/generated/ApiGen_Concepts_PropertyActivity';
+import { ApiGen_Concepts_PropertyActivityInvoice } from '@/models/api/generated/ApiGen_Concepts_PropertyActivityInvoice';
+import { ApiGen_Concepts_PropertyActivityInvolvedParty } from '@/models/api/generated/ApiGen_Concepts_PropertyActivityInvolvedParty';
+import { ApiGen_Concepts_PropertyActivityProperty } from '@/models/api/generated/ApiGen_Concepts_PropertyActivityProperty';
+import { ApiGen_Concepts_PropertyMinistryContact } from '@/models/api/generated/ApiGen_Concepts_PropertyMinistryContact';
+import { getEmptyBaseAudit } from '@/models/defaultInitializers';
+import { exists, isValidIsoDateTime } from '@/utils';
+import { emptyStringtoNullable, toTypeCodeNullable } from '@/utils/formUtils';
+
+export class ManagementActivityFormModel {
+  activityTypeCode = '';
+  activitySubtypeCode = '';
+  activityStatusCode = '';
+  requestedDate = '';
+  completionDate = '';
+  description = '';
+  requestedSource = '';
+  ministryContacts: (IContactSearchResult | null)[] = [null];
+  involvedParties: (IContactSearchResult | null)[] = [null];
+  serviceProvider: IContactSearchResult | null = null;
+  invoices: ActivityInvoiceFormModel[] = [];
+  pretaxAmount = 0;
+  gstAmount = 0;
+  pstAmount = 0;
+  totalAmount = 0;
+  activityProperties: ActivityPropertyFormModel[] = [];
+
+  constructor(
+    readonly id: number | null = null,
+    readonly managementFileId: number | null = null,
+    readonly rowVersion: number | null = null,
+  ) {
+    this.id = id;
+    this.managementFileId = managementFileId;
+    this.rowVersion = rowVersion;
+  }
+
+  toApi(): ApiGen_Concepts_PropertyActivity {
+    const apiActivity: ApiGen_Concepts_PropertyActivity = {
+      id: this.id ?? 0,
+      managementFileId: this.managementFileId,
+      activityTypeCode: toTypeCodeNullable(this.activityTypeCode),
+      activitySubtypeCode: toTypeCodeNullable(this.activitySubtypeCode),
+      activityStatusTypeCode: toTypeCodeNullable(this.activityStatusCode),
+      requestAddedDateOnly: this.requestedDate,
+      completionDateOnly: emptyStringtoNullable(this.completionDate),
+      description: this.description,
+      requestSource: this.requestedSource,
+
+      isDisabled: false,
+      serviceProviderOrgId: this.serviceProvider?.organizationId ?? null,
+      serviceProviderOrg: null,
+      serviceProviderPersonId: this.serviceProvider?.personId ?? null,
+      serviceProviderPerson: null,
+
+      involvedParties: this.involvedParties
+        .filter(exists)
+        .map<ApiGen_Concepts_PropertyActivityInvolvedParty>(x => {
+          return {
+            id: 0,
```

Comment:
hmm, not sure about this, I think this will may ef to add/remove an entity every time this is saved even if no changes are made to the involved parties.

Reply:
This was also lifted from the original activity implementation. We can enhance it for sure but it works for creation. For update (another upcoming story) I agree would be better to change

Reply:
I will change it to use the existing ID (if is there, otherwise use 0)

Reply:
After looking into this. in the **fromApi()** method we "drop" those IDs because we convert to an IContactSearchResult so there is no ID to use back in the **toApi** method.

Given that this code was already in place in ad-hoc activity I would lean to leave it as-is and discuss a better pattern in our dev meeting

Reply:
That seems fine - my comment about copying code, is that copied code needs to follow our current standards, not the standards that code was developed with. We don't want to pervade negative patterns from old code if there is a viable solution.